### ISSUE TITLE
Добавить вывод вермени и даты в лог

### ISSUE THREAD

**Issue by Khorevaa** [oscript-library/logos#6]
Сейчас лог неинформативен при просмотре, постфактум.

Предлагю добавить вывод даты и времени, можно по параметры или настройке

**Comment by artbear** [comment]
Да, нужно указывать параметром.
Некоторые CI-серверы при выводе лога сами указывают время - ТимСити, например.

**Comment by Stepa86** [comment]
Вывод даты и времени легко делается через свою раскладку. Смотри пример у меня https://github.com/Stepa86/oScript-Examples/blob/master/src/CopyBase.os#L95 и собственно сам класс раскладки https://github.com/Stepa86/oScript-Examples/blob/master/src/%D0%9A%D0%BB%D0%B0%D1%81%D1%81%D1%8B/%D0%97%D0%B0%D0%BC%D0%B5%D1%80.os . В том числе через раскладку я сделал иерархичность вывода

**Comment by artbear** [comment]
Да, мы также в нескольких скриптах делали через свою раскладку.
Класс раскладки и был добавлен для возможности динамического переопределения вывода.

Но параметризовать все равно полезно именно в логосе, чтобы уменьшить дублирование кода в скриптах.

**Comment by EvilBeaver** [comment]
В логос надо добавить возможность менять раскладку через конфигурационные
параметры. Как и в случае с уровнями. Посмотрите в lo4j

17 Фев 2017 г. 9:43 пользователь "Artur Ayukhanov" <notifications@github.com>
написал:

**Comment by artbear** [comment]
>В логос надо добавить возможность менять раскладку через конфигурационные
параметры. Как и в случае с уровнями. Посмотрите в lo4j

@EvilBeaver Создай, плиз, новый ишуз.

**Comment by dmpas** [comment]
@Khorevaa @Stepa86 @EvilBeaver @artbear  освежим в памяти, что делаем с этой задачкой?

**Comment by EvilBeaver** [comment]
Упрощаем создание форматеров.

**Comment by nixel2007** [comment]
запишу, чтобы не потерялось.

1. в log4j у форматеров есть параметры, которые можно устанавливать, в т.ч. через конфигурационные файлы
2. один из основных форматеров (PatternLayout) имеет несколько встроенных подстановок для эээ аналога СтрШаблон. типа %t - это имя лога, %d - текущая дата (с возможностью определения формата вывода). Форматная строка может выглядеть, например, так `%-5p [%t]: %m%n` - вывод уровня лога (в 5 символов), имя лога в квадратных скобках, само сообщение и перевод строки. более подробно (старое апи) - https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/PatternLayout.html. новое апи https://logging.apache.org/log4j/2.x/manual/layouts.html
